drivers/text_display/text_display_noritake_itron.c

```diff
@@ -0,0 +1,259 @@
+#define NORITAKE_ITRON_CMD_DISPLAY_CLEAR 0x0c
+//#define NORITAKE_ITRON_CMD_HOME 0x0b
+#define NORITAKE_ITRON_CMD_CURSOR 0x43
+#define NORITAKE_ITRON_CMD_ACTION 0x28
+#define NORITAKE_ITRON_CMD_N 0x61
+#define NORITAKE_ITRON_CMD_SCREEN_SAVER 0x40
+
+//.h file
+#define NORITAKE_ITRON_BRIGHTNESS_MIN 1
+#define NORITAKE_ITRON_BRIGHTNESS_MAX 8
+
+#define NORITAKE_ITRON_CHARACTER_MIN 0x20
+#define NORITAKE_ITRON_CHARACTER_BACK_SPACE 0x08
+#define NORITAKE_ITRON_CHARACTER_TAB 0x09
+#define NORITAKE_ITRON_CHARACTER_LINE_FEED 0x0a
+#define NORITAKE_ITRON_CHARACTER_CARRIAGE_RETURN 0x0d
+
+
+struct text_display_noritake_itron_data {
+	uint8_t unused;
+};
+
+struct text_display_noritake_itron_config {
+	const struct device *uart;
+	uint16_t columns;
+	uint16_t rows;
+	struct gpio_dt_spec reset_gpio;
+	struct gpio_dt_spec busy_gpio;
+};
+
+static void send_cmd(const struct device *uart, const uint8_t *command, uint8_t length);
+
+static int noritake_itron_init(const struct device *dev)
+{
+	const struct text_display_noritake_itron_config *config = dev->config;
+	struct text_display_noritake_itron_data *data = dev->data;
+	int rc;
+
+	if (!device_is_ready(config->uart)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	/* Configure and set busy GPIO */
+	if (config->busy_gpio.port) {
+		rc = gpio_pin_configure_dt(&config->busy_gpio, GPIO_INPUT);
+		if (rc < 0) {
+			LOG_ERR("Configuration of text display busy GPIO failed");
+			return rc;
+		}
+	}
+
+	/* Configure and set reset GPIO */
+	if (config->reset_gpio.port) {
+		rc = gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_INACTIVE);
+		if (rc < 0) {
+			LOG_ERR("Configuration of text display reset GPIO failed");
+			return rc;
+		}
+	}
+
+	return 0;
+}
+
+static int noritake_itron_power_set(const struct device *dev, bool enabled)
+{
+	const struct text_display_noritake_itron_config *config = dev->config;
+	uint8_t cmd[] = {NORITAKE_ITRON_CMD_USER_SETTING, NORITAKE_ITRON_CMD_ACTION,
+			 NORITAKE_ITRON_CMD_N, NORITAKE_ITRON_CMD_SCREEN_SAVER, (uint8_t)enabled};
+
+	send_cmd(config->uart, cmd, sizeof(cmd));
+
+	return 0;
+}
+
+static int noritake_itron_cursor_set(const struct device *dev, bool enabled)
+{
+	const struct text_display_noritake_itron_config *config = dev->config;
+	uint8_t cmd[] = {NORITAKE_ITRON_CMD_USER_SETTING, NORITAKE_ITRON_CMD_CURSOR,
+			 (uint8_t)enabled};
+
+	send_cmd(config->uart, cmd, sizeof(cmd));
+
+	return 0;
+}
+
+static int noritake_itron_position_blinking_set(const struct device *dev, bool enabled)
+{
+	return -ENOTSUP;
+}
+
+static int noritake_itron_cursor_position_set(const struct device *dev, uint8_t type, uint8_t x, uint8_t y)
+{
+	return -ENOTSUP;
+}
+
+static int noritake_itron_display_position_set(const struct device *dev, uint8_t type, uint8_t x, uint8_t y)
+{
+	return -ENOTSUP;
+}
+
+static int noritake_itron_details_get(const struct device *dev, uint16_t *columns, uint16_t *rows, uint8_t *character_set)
+{
+	return -ENOTSUP;
+}
```

Comment:
if APIs are optional, maybe allow them to be `NULL` and return `-ENOSYS` at interface level.